charts/rancher-istio/rancher-istio/100.0.0/charts/kiali/templates/_helpers.tpl

```diff
@@ -50,8 +50,20 @@ app.kubernetes.io/part-of: "kiali"
 Selector labels
 */}}
 {{- define "kiali-server.selectorLabels" -}}
+{{- $releaseName := .Release.Name -}}
+{{- $fullName := include "kiali-server.fullname" . -}}
 app.kubernetes.io/name: kiali
-app.kubernetes.io/instance: {{ include "kiali-server.fullname" . }}
+{{- if .Release.IsUpgrade }}
+    {{- $labels := (lookup "apps/v1" "Deployment" .Release.Namespace $fullName).metadata.labels }}
```

Comment:
What happens if the lookup fails? 

e.g. the user deletes the Deployment after deploying and triggers an upgrade.

I believe the current code would cause an error on trying to render the template on an upgrade right?

I think we should modify this logic slightly to say something like:

```
$label := $fullName
lookup ...
if isUpgrade and resource exists:
  get label value
  if label value is not nil
    set $label to that value
  end
end
app.kubernetes.io/instance: {{ $label }}
```

Reply:
If Helm supports preseving omitted labels automatically, it would be nice if we can drop the stuff in the top-level if above entirely.